### PR TITLE
Update sqlite from 3.38.2 to 3.38.3

### PR DESCRIPTION
**Pull request by cbouss**
# Changes
- Update version from `3.38.2` to `3.38.3`.
- Add license_family, update dev_url
- Remove unused patch file

# Review Information
Source: https://sqlite.org/src/dir?ci=trunk
Change log: https://sqlite.org/src/timeline?from=branch-3.38&to=version-3.38.2
License: https://sqlite.org/src/file?name=LICENSE.md&ci=trunk
No current vulnerabilities: https://www.sqlite.org/cves.html#cvetab

# Testing
Existing tests passed.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ package:
 
 source:
   url: https://www.sqlite.org/{{ year }}/sqlite-autoconf-{{ version_coded }}.tar.gz
-  sha256: e7974aa1430bad690a5e9f79a6ee5c8492ada8269dc675875ad0fb747d7cada4
+  sha256: 61f2dd93a2e38c33468b7125967c3218bf9f4dd8365def6025e314f905dc942e
   patches:
     - expose_symbols.patch  # [win]
```
Comment (dholth) [inline recipe/meta.yaml:17]: So the 0001- patch is not used, and a similar non-numbered patch is used?

Comment (cbouss) [reply recipe/meta.yaml:17]: Correct, that's a change coming from conda forge on a previous release.
It's the same patch, however. Before it used to be applied to all architecture and contained #ifdef _WIN32.
Now it is only applied to windows and does not contain #ifdef _WIN32.

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -56,13 +55,14 @@ about:
   home: https://www.sqlite.org/
   license: Unlicense
```
Comment (ScottBailey) [inline recipe/meta.yaml]: I believe license should be `blessing` and `license_url` should be removed in favor of `license_file` with a value of `LICENSE.md`. The license can be found here: https://sqlite.org/src/file?name=LICENSE.md&ci=trunk

**Review by ScottBailey — CHANGES_REQUESTED** [review]
Update license info and then looks good.

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: I left unlicense which is more explicit imho, and added LICENSE.md.

**Review by ScottBailey — APPROVED** [review]

**Pull request merged**